app/controllers/locations_controller.rb

```diff
@@ -1,6 +1,10 @@
 class LocationsController < ApplicationController
   def index
-    @locations = Location.all
+    @locations = if params[:search]
+                   Location.search(params[:search])
+                 else
+                   Location.all
+                 end
   end
 
   def new
```

Comment:
Fun fact: this looks like a perfect job for the Ruby ternary operator. `@locations = params[:search] ? Location.search(params[:search]) : Location.all`